File: problems/2239-find-closest-number-to-zero/solution.java
class Solution {
    public int findClosestNumber(int[] nums) {
        int closest = nums[0];
        int minDiff = Math.abs(nums[0]);

        for (int i = 1; i < nums.length; i++) {
            int diff = Math.abs(nums[i]);

            if (diff < minDiff) {
                minDiff = diff;
                closest = nums[i];
            } else if (diff == minDiff) {
                closest = Math.max(closest, nums[i]);
            }
        }

        return closest;
    }
}

File: problems/2239-find-closest-number-to-zero/solution.py
class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        closest = nums[0]
        min_diff = abs(nums[0])
        for num in nums:
            diff = abs(num)
            if diff < min_diff:
                min_diff = diff
                closest = num
            elif diff == min_diff:
                closest = max(closest, num)
        return closest